subsys/bluetooth/host/l2cap.c

```diff
@@ -1407,6 +1412,10 @@ static void le_ecred_reconf_req(struct bt_l2cap *l2cap, uint8_t ident,
 response:
 	buf = l2cap_create_le_sig_pdu(BT_L2CAP_ECRED_RECONF_RSP, ident,
 				      sizeof(*rsp));
+	if (!buf) {
+		LOG_DBG("Unable to respond to reconf request");
+		return;
+	}
```

Comment:
What happens if we don't send the response? Would the L2CAP channel break? 

Wondering if this should be a `LOG_WRN` or `LOG_ERR`, or maybe even trigger some other behavior? A retry?

Reply:
it will eventually timeout (RTX timeout it's called in the code). I think it's configured to two seconds.

Reply:
What happens after the timeout? Disconnect?

If we know that will happen, why not have it disconnect earlier? Or can we not do that because we do not have buffers to do that?

Reply:
Either way, this should really be a `LOG_ERR()`. Or `LOG_WRN()` if you think this error can be caused by a misbehaving peer rather than a local issue (e.g. suboptimal buf count)

Reply:
I made a `DBG` because I don't think it can happen because of a misbehaving peer. If it happens, it will most likely be not enough buffers. Also to keep flash usage low and all that.

Reply:
My thought process is that the channel will time out, a warning will be printed. That prompts the developer to enable `DBG` logging and then find out that a buffer alloc fails.
But I'm okay with promoting it to a `WRN` if you both think it's the right level

Reply:
I'd prefer a `LOG_WRN`

Reply:
I get the feeling that we should have some documented guidelines on how to choose between DBG/WRN/ERR/INF. The most simple approach, which I'd be in favor of (and have used in most of the Bluetooth subsystem code):
 - `ERR` is when something failed locally (the situation here)
 - `WRN` is when something that's not necessarily a local failure, but might be of concern (this could be e.g. when receiving something invalid from a remote peer)
 - `DBG` is for other messages that may be helpful for debugging
 - `INF` should generally be avoided. It's used in a couple of places, like when initializing the host and logging some essential information that was discovered about the controller (e.g. which core spec version it supports).

Reply:
@jori-nordic if you think this is more of a case of a failure, but which doesn't prevent proceeding with normal operation, then `DBG` could also be justified, but I'd accompany it with a code comment, since otherwise it just looks wrong to see that in a failure branch.

Reply:
(Just saving that list for the coding guidelines :+1: )
`<rant>`
For the "it just looks wrong". Please take a look at the rest of the host, it's cases on cases on cases of exactly this, just silently noping out. At least I left a debug message.
`</rant>`

I think I'll promote it to a warning though:

If ATT is the user (most likely for ecred), that error won't prevent operation, as in that case the fixed ATT channel will still be open. But that EATT bearer will time-out and be rendered useless.

If the application's the user, then they won't have any other signal than an eventual time-out. The time-out is not too long, but still annoying. That's why I'm promoting it to `WRN`.

Reply:
> For the "it just looks wrong". Please take a look at the rest of the host, it's cases on cases on cases of exactly this, just silently noping out. At least I left a debug message.

Ah, not having any log statement at all is a fifth option that we didn't yet cover. There may indeed be places that should have a log message, but in many cases either the layer below or above cover for the lack of them. However, especially if you don't pass up any error value (e.g. `void` return) then a log message is the only way to know that the branch was taken.

Btw, I thought we were talking about both of your added log messages here, but I see you left the other one as `LOG_DBG()`. Wouldn't it be better to be consistent?

Reply:
> Btw, I thought we were talking about both of your added log messages here, but I see you left the other one as LOG_DBG(). Wouldn't it be better to be consistent?

I think the current behavior is correct. In the first one with `LOG_DBG` we are able to reject a request, and thus does not cause timeouts or any other "odd" behavior.

Reply:
> > Btw, I thought we were talking about both of your added log messages here, but I see you left the other one as LOG_DBG(). Wouldn't it be better to be consistent?
> 
> I think the current behavior is correct. In the first one with `LOG_DBG` we are able to reject a request, and thus does not cause timeouts or any other "odd" behavior.

Alright. Fair enough.

Reply:
> > > Btw, I thought we were talking about both of your added log messages here, but I see you left the other one as LOG_DBG(). Wouldn't it be better to be consistent?
> > 
> > 
> > I think the current behavior is correct. In the first one with `LOG_DBG` we are able to reject a request, and thus does not cause timeouts or any other "odd" behavior.
> 
> Alright. Fair enough.

@Thalley just to clarify, by "reject" do you mean that we send a rejection PDU to the peer? Maybe I'm not reading the code right, but I can't find where that'd happen as a consequence of the `-ENOMEM` return. There are two places calling the function, neither one seems to result in a rejection PDU and the other one just plainly ignores the return value completely.

Reply:
It's called by this public function:

```c
int bt_l2cap_ecred_chan_connect(struct bt_conn *conn,
				struct bt_l2cap_chan **chan, uint16_t psm)
{
	int i, err;

	LOG_DBG("conn %p chan %p psm 0x%04x", conn, chan, psm);

	if (!conn || !chan) {
		return -EINVAL;
	}

	/* Init non-null channels */
	for (i = 0; i < L2CAP_ECRED_CHAN_MAX_PER_REQ; i++) {
		if (!chan[i]) {
			break;
		}

		err = l2cap_ecred_init(conn, BT_L2CAP_LE_CHAN(chan[i]), psm);
		if (err < 0) {
			i--;
			goto fail;
		}
	}

	return l2cap_ecred_conn_req(chan, i);
```

So if `bt_l2cap_ecred_chan_connect` gets called by the application and we do not have enough buffers to send the request, the `ENOMEM` is returned to the caller. 

We do have a case where the return value of `l2cap_ecred_conn_req` is not used though :(

Reply:
But the LOG statement is actually a bit weird, as it is not a response we are sending, but a request.